src/scripts.js

```diff
@@ -1,47 +1,91 @@
-// https://www.youtube.com/watch?v=_3oSWwapPKQ
-
-// Examples
 // scripts 👇
-// import ActivityRepository from './ActivityRepository';
-// import {
-//   displayAllData,
-//   weeksWaterInput,
-//   displayUserSelectWeek,
-// } from './domUpdates';
-import getAllData from './apiCalls';
-
-// An example of how you tell webpack to use a CSS (SCSS) file - look at @use here instead of import - must impost ALL scss files here
+import { fetchAllData } from './apiCalls';
+import * as dayjs from 'dayjs';
+import Trip from './Trip';
+import Traveler from './Traveler';
+import {
+  renderDestinationsGrid,
+  renderTripsGrid,
+  renderCurrentTrip,
+  displayBookTripPage,
+  displayYearToDateSpent,
+  getTripEstimate,
+  displayTravelerDashBoard,
+} from './domUpdates';
+
 // styling 👇
 import './css/base.scss';
 import './css/_reset.scss';
 import './css/_variables.scss';
-// example of @use syntax - didn't work in fitlit
-// @use ‘_colors.scss’ as *;
-
-// An example of how you tell webpack to use an image (also need to link to it in the index.html - example in html)
-// images 👇
-import './images/turing-logo.png';
 
-// An example of exports to the dom file (after all imports put the exports)
-// exports 👇
-// export { currentUser, userRepo, hydrationStats, sleepStats, activityStats };
-
-console.log('This is the JavaScript entry file - your code begins here.');
+// query selectors 👇
+const tripsButtons = document.getElementById('aside');
+export const mainDisplay = document.getElementById('main');
+export const bookTripBtn = document.getElementById('bookNow');
+const getEstimateBtn = document.getElementById('getEstimate');
 
 // global variables 👇
+export let dateToday = dayjs().format('YYYY/MM/DD');
+export let currentTraveler;
+export let trips;
+export let destinationsData;
+export let tripsData;
 
 // event listeners 👇
 
+// display on load 👇
+
 // event handlers and functions 👇
-getAllData().then((data) => {
-  console.log(data[2]); // instantiate all destinations to display on dom for the user
-  console.log(data[0]); // build helper function to instantiate/get user
-  console.log(data[1]); // build helper function to filter through array to instantiate/get all users trips // will eventually need to filter by date for past/present/future/pending and have those arrays on the user class
-});
+const startApp = () => {
+  fetchAllData()
+    .then((data) => {
+      let travelerData = data[0];
+      tripsData = data[1].trips;
+      destinationsData = data[2].destinations;
+      getTrips(tripsData, travelerData, destinationsData);
+      getTraveler(travelerData, trips);
+      console.log('in fetch data');
+    })
+    .then(displayDashboard);
+};
+
+const getTrips = (tripsData, travelerData, destinationsData) => {
+  trips = tripsData
+    .filter((trip) => trip.userID === travelerData.id)
+    .map((trip) => {
+      const instantiatedTrip = new Trip(trip, destinationsData);
+      return instantiatedTrip;
```

Comment:
Not sure if you need the word "instantiated" in this variable since new Trip makes it clear it is instantiating a new instance of trip. What do you think?

Reply:
Absolutely